scenarios/csharp/GitHubLovesSparkDotNet/Program.cs

```diff
@@ -0,0 +1,128 @@
+{
+    public class Program
+    {
+        #region Cloud Run
+        public const string StorageConfigKey
+            = "fs.azure.account.key.<your-storage-account-name>.dfs.core.windows.net";
+        public const string SecureStorageKey = "<your-storage-account-key>";
+        public const string CloudStoragePath
+            = "abfss://<filesystem>@<storage-account>.dfs.core.windows.net/<path>/";
+        #endregion
+
+        #region Local Run
+        public static string LocalDataStoragePath { get; } =
+            Path.Combine(
+                Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location),
+                "Resources" + Path.DirectorySeparatorChar); 
+        #endregion
+
+        static void Main(string[] args)
+        {
+            if (args.Length == 0)
+            {
+                PrintUsage();
+                return;
+            }
+
+            string StoragePath = (args[0] == "local") ? LocalDataStoragePath : CloudStoragePath;
+
+            Builder sparkBuilder = SparkSession.Builder();
+
+            if (args[0] == "cloud")
+                sparkBuilder = sparkBuilder.Config(StorageConfigKey, SecureStorageKey);
+
+            SparkSession spark = sparkBuilder
+                    .AppName(@"Github 💖 .NET for Apache Spark")
+                    .GetOrCreate();
+
+            // Initialize all dataframes (which point to CSV files on the storage system)
+            DataFrame projects = spark
+                    .Read()
+                    .Option("header", "true")
+                    .Schema("id INT, url STRING, owner_id INT, name STRING, " +
+                            "descriptor STRING, language STRING, created_at STRING, " +
+                            "forked_from INT, deleted STRING, updated_at STRING")
+                    .Csv(StoragePath + "projects.csv")
+                    .Filter(Col("language") == "C#");
+
+            DataFrame watchers = spark
+                    .Read()
+                    .Option("header", "true")
+                    .Schema("repo_id INT, user_id INT, created_at TIMESTAMP")
+                    .Csv(StoragePath + "watchers.csv");
+
+            DataFrame commits = spark
+                    .Read()
+                    .Option("header", "true")
+                    .Schema("id INT, sha STRING, author_id INT, committer_id INT, " +
+                            "project_id INT, created_at TIMESTAMP")
+                    .Csv(StoragePath + "commits.csv");
+
+            // Use functional programming to find top C# projects by stars
+            DataFrame stars = projects
+              .Join(watchers, Col("id") == watchers["repo_id"])
+              .GroupBy("name")
+              .Agg(Count("*").Alias("stars"))
+              .OrderBy(Desc("stars"));
+
+            stars.Show();
+
+            // Or... Spark SQL!!!
+            projects.CreateOrReplaceTempView("projects");
+            watchers.CreateOrReplaceTempView("watchers");
+            spark.Sql(@"
```

Comment:
What is the "@" for? I thought it signified parameters in SQL, but why does this query need it?

Reply:
Oh, it is a verbatim string literal. [more info](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/tokens/verbatim)